File: SoftRhUsuariosBO/src/pe/edu/pucp/softrh/usuarios/bo/CuponBO.java
package pe.edu.pucp.softrh.usuarios.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softrh.usuarios.dao.CuponDAO;
import pe.edu.pucp.softrh.usuarios.daoimp.CuponDAOImp;
import pe.edu.pucp.softrh.usuarios.model.Cupon;
import pe.edu.pucp.softrh.usuarios.model.Trabajador;

public class CuponBO {
	private CuponDAO cuponDAO;

	public CuponBO() {
		this.cuponDAO = new CuponDAOImp();
	}

	public Integer insertar(String codigo, String descripcion, Date fechaInicio, Date fechaFin, Trabajador trabajador) {
		Cupon cupon = new Cupon(codigo, descripcion, fechaInicio, fechaFin, trabajador);
		cupon.setIdCupon(cuponDAO.insertar(cupon));
		return cupon.getIdCupon();
	}

	public Integer modificar(Integer idCupon, String codigo, String descripcion, Date fechaInicio, Date fechaFin, Trabajador trabajador) {
		Cupon cupon = new Cupon(codigo, descripcion, fechaInicio, fechaFin, trabajador);
		cupon.setIdCupon(idCupon);
		return cuponDAO.modificar(cupon);
	}

	public Integer eliminar(Integer idCupon) {
		return cuponDAO.eliminar(idCupon);
	}

	public ArrayList<Cupon> listarTodos() {
		return cuponDAO.listarTodos();
	}

	public Cupon obtenerPorId(Integer idCupon) {
		return cuponDAO.obtenerPorId(idCupon);
	}
}

File: SoftRhWeb/SoftRhUsuariosBO/CuponBO.cs
using SoftRhBaseBO;
using SoftRhBaseBO.Servicios;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftRhUsuariosBO
{
	public class CuponBO : BaseBO
	{
		public int insertar(string codigo, string descripcion, DateTime fechaInicio, DateTime fechaFin, trabajador _trabajador)
		{
			return CuponWS.insertarCupon(codigo, descripcion, fechaInicio, fechaFin, _trabajador);
		}

		public int modificar(int idCupon, string codigo, string descripcion, DateTime fechaInicio, DateTime fechaFin, trabajador _trabajador)
		{
			return CuponWS.modificarCupon(idCupon, codigo, descripcion, fechaInicio, fechaFin, _trabajador);
		}

		public int eliminar(int idCupon)
		{
			return CuponWS.eliminarCupon(idCupon);
		}

		public BindingList<cupon> listarTodos()
		{
			return new BindingList<cupon>(CuponWS.listarCuponesTodos());
		}

		public cupon obtenerPorId(int idCupon)
		{
			return CuponWS.obtenerCuponPorId(idCupon);
		}
	}
}
